be/src/exec/es/es_scan_reader.cpp

```diff
@@ -155,7 +183,9 @@ Status ESScanReader::get_next(bool* scan_eos, std::unique_ptr<ScrollParser>& scr
         if (_exactly_once) {
             return Status::OK();
         }
-        RETURN_IF_ERROR(_network_client.init(_next_scroll_url));
+        // we do not enable set_fail_on_error for ES http request to get more detail error messages
+        bool set_fail_on_error = false;
+        RETURN_IF_ERROR(_network_client.init(_next_scroll_url, set_fail_on_error));
```

Comment:
Why change error handling behavior here?

Reply:
When the http response of ES is not ok, it will provide a specific error hint detail in response body.
That will help us pinpoint where the problem is.
For example:
```
curl localhost:29200/test2/_search?pretty -H'Content-Type:application/json' -d '{"match:}'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "json_e_o_f_exception",
        "reason" : "Unexpected end-of-input in field name\n at [Source: (org.elasticsearch.common.io.stream.ByteBufferStreamInput); line: 1, column: 10]"
      }
    ],
    "type" : "json_e_o_f_exception",
    "reason" : "Unexpected end-of-input in field name\n at [Source: (org.elasticsearch.common.io.stream.ByteBufferStreamInput); line: 1, column: 10]"
  },
  "status" : 400
}
```
We disable `set_fail_on_error` try to log the response body.